File: backend/src/main/java/org/jdk/project/service/statistics/StatisticsQuerySupport.java
package org.jdk.project.service.statistics;

final class StatisticsQuerySupport {

  private StatisticsQuerySupport() {}

  static String trimToNull(String value) {
    if (value == null || value.trim().isEmpty()) {
      return null;
    }
    return value.trim();
  }

  static String normalizeAccessType(String value) {
    if (value == null) {
      return null;
    }
    return switch (value) {
      case "1" -> "ID_CARD";
      case "2" -> "QRCODE";
      case "3" -> "FACE";
      default -> value;
    };
  }

  static String normalizePassengerStatus(String value) {
    if (value == null) {
      return null;
    }
    return switch (value) {
      case "1" -> "IN";
      case "0" -> "OUT";
      default -> value;
    };
  }
}

File: backend/src/main/kotlin/org/jdk/project/service/statistics/StatisticsQuerySupport.kt
package org.jdk.project.service.statistics

object StatisticsQuerySupport {
    @JvmStatic
    fun trimToNull(value: String?): String? = value?.trim()?.takeIf { it.isNotEmpty() }

    @JvmStatic
    fun normalizeAccessType(value: String?): String? =
        when (value) {
            null -> null
            "1" -> "ID_CARD"
            "2" -> "QRCODE"
            "3" -> "FACE"
            else -> value
        }

    @JvmStatic
    fun normalizePassengerStatus(value: String?): String? =
        when (value) {
            null -> null
            "1" -> "IN"
            "0" -> "OUT"
            else -> value
        }
}
